src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -650,6 +654,84 @@ def process_declarations(self, parent, nodes, arg_list):
                     "Could not process '{0}'. Statement Function declarations "
                     "are not supported.".format(str(stmtfn)))
 
+
+    @staticmethod
+    def _process_kind_selector(symbol_table, type_spec):
+        '''
+        Processes the fparser2 parse tree of the type specification of a
+        variable declaration in order to extract KIND information. This
+        information is used to determine the precision of the variable (as
+        supplied to the Symbol constructor).
+
+        :param symbol_table: the SymbolTable associated with the code \
+                             being processed.
+        :type symbol_table: :py:class:`psyclone.psyGen.SymbolTable`
+        :param type_spec: the fparser2 parse tree of the type specification.
+        :type type_spec: :py:class:`fparser.two.Fortran2003.Intrinsic_Type_Spec
+
+        :returns: the precision associated with the type specification.
+        :rtype: int or :py:class:`psyclone.psyGen.Symbol.Precision` or \
+                :py:class:`psyclone.psyGen.PrecisionSymbol` or NoneType
+
+        '''
+        if not isinstance(type_spec.items[1], Fortran2003.Kind_Selector):
+            return None
+        kind_selector = type_spec.items[1]
+
+        intrinsics = walk_ast(kind_selector.items,
+                              [Fortran2003.Intrinsic_Function_Reference])
+        if intrinsics and \
+           isinstance(intrinsics[0].items[0],
+                      Fortran2003.Intrinsic_Name) and \
+                      str(intrinsics[0].items[0]).lower() == "kind":
+            # We have kind=KIND(literal). items[1] is an
+            # Actual_Arg_Spec_List with the first entry being the
+            # literal.
+            literal = intrinsics[0].items[1].items[0]
+            if isinstance(literal,
+                          Fortran2003.Real_Literal_Constant):
+                if "d0" in str(literal).lower():
```

Comment:
Oh yes, that's a good point. I was writing a frontend for the code I know rather than allowing for the full spectrum :-) I've stared hard at the spec and I think pretty much anything goes :-( Obviously, `kind=KIND(1_idef)` == `kind=idef` so it's pretty pointless. I'll give it some thought.